app/controllers/bank_accounts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BankAccountsController < ApplicationController
+  def index
+    if current_user.banker ==true || current_user.staff == true
+      @accounts = BankAccount.all
+      characters = Character.all
+      @characters_without_accounts = []
+      characters.each do |character|
+        if !BankAccount.has_account?(character.id) == true && character.archived==false
+          @characters_without_accounts << character
+        end
+      end
+      @account = BankAccount.new
+    else
+      authenticate_staff!
+    end
+  end
+
+  def show
+    @character = Character.find_by(slug: params[:character_id])
+      if !@character
+        @character = Character.find(params[:character_id])
+      end
```

Comment:
Syntactically, I'd prefer to clean this up, and test for the positive case first. You can do that by using `if @character.present?`, instead of `if !@character` (which just doesn't read cleanly, for my tastes.)

Reply:
But if I do it that way, and the character is present wouldn't I have to put the @account and @transactions inside the conditional, then if the character isn't present have it try to find the character a second way, then again set the @account and @transactions?

Reply:
My head hurts. Why are we ever finding characters by anything other than `slug`, again? We've overridden the `to_param` method on that model, does this ever actually come up, in practice?

Reply:
Mostly cause I don't know how slug works, and when I got to the bank account path I get http://localhost:3000/characters/6/bank_accounts/3.

Reply:
I suspect it's because you've allowed yourself to develop the bad habit of passing the `id` to the `path` helper, instead of the entire object.

So for instance if you have a resource called `somethings`, and you have a particular instance of that resource assigned to the variable `@something`, you have your ERB set up so that you're linking to, e.g. `something_path(@something.id)`.

This is bad and wrong.

If you instead link to `something_path(@something)`, ActiveRecord calls `to_param` on the `Something` model. The default behavior for ActiveRecord models is for `to_param` to return the `id`. But there are all sorts of security reasons not to expose sequential ids, so for a number of our models (incl. characters) the `to_param` method has been overridden, to provide a *different* unique identifier (in this case, a `slug`, which is just a catch-all term for a parameterized version of the name of something, e.g. the slug for 'Pat Whelan' would be 'pat-whelan'.)